### PR TITLE
[agent] remove wpantund support

### PR DESCRIPTION
**Pull request by bukepo**
* Remove wpantund support
* Fix building on macOS

### CODE REVIEW

**Comment by bukepo** [comment]
This PR depends on https://github.com/openthread/openthread/pull/4664.

**Comment by bukepo** [comment]
@jwhui since this requires document update on openthread.io, but there will be several changes before we can update document. So I created a temporary branch 0.02.00/develop as the target branch so that we can review one by one and merge that branch when everything is ready.

**Comment by bukepo** [comment]
Ping~

File: Android.mk [hunk 1]
```diff
@@ -126,28 +127,19 @@ LOCAL_SRC_FILES := \
     src/agent/agent_instance.cpp \
     src/agent/border_agent.cpp \
     src/agent/main.cpp \
-    src/common/event_emitter.cpp \
-    src/common/logging.cpp \
-    src/utils/hex.cpp \
-    src/utils/strcpy_utils.cpp \
-    $(NULL)
-
-ifeq ($(USE_OTBR_DAEMON), 1)
```
Comment (jwhui) [inline Android.mk:135]: Should we completely remove all if-statements on `USE_OTBR_DAEMON`?

**Pull request merged**